### DISCUSSION TITLE
How I can make println command output to console?

### DISCUSSION THREAD

**Discussion by moonvent** [mrcjkb/rustaceanvim#222] [Q&A] [answered]
Hi guys! I'm new one! And start to learn Rust, and today I try to add this to my nvim, and now I can run and debug, but can't see the output from println function, why? Setup is standart:
```
{
  'mrcjkb/rustaceanvim',
  version = '^4', -- Recommended
  ft = { 'rust' },
}
``` 
Result is:

![image](https://github.com/mrcjkb/rustaceanvim/assets/51057725/abcd4f9e-90e0-467b-890a-4f1b51bb830d)

I want to get the output from println commands =)

**Comment by mrcjkb** [comment]
Hey :wave: 

You will probably want to look at the [nvim-dap `:h dap-widgets` API documentation](https://github.com/mfussenegger/nvim-dap/blob/master/doc/dap.txt#widget-api).

Or, you can use [nvim-dap-ui](https://github.com/rcarriga/nvim-dap-ui?tab=readme-ov-file).

**Reply by moonvent** [reply]
Unfortunately, I just switch to vanila nvim dap rust, and make it throught this =(

**Reply by mrcjkb** [reply]
I don't understand this comment.

**Reply by moonvent** [reply, marked as answer]
Unfortunately, I just switched back to the old stack, LSP + DAP, because many things there make more sense, such as the key bindings are already set up and they are more comprehensive, + I tried to solve a problem for a couple of hours but couldn't manage it, and eventually decided to just switch technologies and that's it

**Reply by jdtaylor311** [reply]
So I found something that may help with then. I used just print statements to look at the values of stdio for the configurations that have been setup. It showed them going to files in what I can only assume is a temp area. I think my print statement was something like `Lua local dap = require('dap') print(vim.inspect(dap.configurations.rust))` this is where I saw the stdio redirect happening. once I set that to a tty I saw the output going there instead. I also had a hunch using the codelldb documentation that if I set the `dap.configurations.rust[x].terminal = 'integrated' `that it would start sending the stdio to the dap console instead which turned out to be the case. Is there anyway to add a way to add the configurations settings that can be joined with the generated config settings to include the terminal option? I'm trying to learn Lua and I would like to do it myself but I'm not sure how I would do it just yet. maybe I could get some pointers and I could submit my first PR? and if that should even be the case or if there might be a better way to do it locally? im still new to neovim and Lua so just trying to understand the options.

**Reply by mrcjkb** [reply]
I've just reverted the commit that is responsible for this. There shouldn't be a need to redirect stdout to a temp file.
